include/boost/iostreams/filter/gzip.hpp

```diff
@@ -505,7 +505,7 @@ class basic_gzip_decompressor : basic_zlib_decompressor<Alloc> {
                 if (footer_.done()) {
                     if (footer_.crc() != this->crc())
                         boost::throw_exception(gzip_error(gzip::bad_crc));
-                    int c = boost::iostreams::get(peek);
+                    c = boost::iostreams::get(peek);
```

Comment:
Is this really needed? You have not touched other `int c`, and I do not see a declaration of the `c` in the outer scope.